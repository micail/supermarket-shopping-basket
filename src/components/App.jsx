import React from 'react';
import { connect } from 'react-redux';
import Basket from './Basket';
import Receipt from './Receipt';


const App = () => {
  return (
    <div className="App container">
      <div className="row">
        <Basket />
        <Receipt />
      </div>
    </div>
  );
};

export const mapStateToProps = (state) => ({
  priceList: state.priceList,
  receipt: state.receipt,
  totals: state.totals,
});

export default connect(mapStateToProps)(App);
