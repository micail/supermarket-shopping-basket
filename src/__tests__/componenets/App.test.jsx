import React from 'react';
import { shallow } from 'enzyme';

import App from '../../components/App';

describe('<App />', () => {
  const wrapper = shallow(<App />);

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
