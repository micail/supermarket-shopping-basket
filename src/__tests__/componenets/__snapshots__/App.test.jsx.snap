// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(App)>
    <App
      dispatch={[Function]}
      priceList={Array []}
      receipt={Array []}
      totals={Array []}
    >
      <div
        className="App"
      >
        <Basket>
          <div
            className="Basket"
          />
        </Basket>
        <Receipt>
          <ul
            className="Receipt"
          />
          <Totals>
            <div
              className="Totals"
            >
              <span>
                -----
              </span>
              <br />
              <span
                className="label"
              >
                Sub-total
              </span>
              <span
                className="value sub-total"
              />
              <br />
              <br />
              <span>
                -----
              </span>
              <br />
              <span
                className="label"
              >
                Total savings
              </span>
              <span
                className="value total-savings"
              />
              <br />
              <span>
                -------------------------
              </span>
              <br />
              <span
                className="label"
              >
                Total to Pay
              </span>
              <span
                className="value total-to-pay"
              />
            </div>
          </Totals>
        </Receipt>
      </div>
    </App>
  </Connect(App)>
</Provider>
`;
